### PR TITLE
fix: Fixing button padding to get correct sizes from design spec

### PR DESCRIPTION
**Pull request by khmakoto**
## Previous Behavior

Once we made the change to calculate the height of the `Button` components automatically by their `content+padding+border` instead of by applying the `height` property, it seems that we missed the mark slightly:
- Small buttons should have a default height of 24px 
- Medium buttons should have a default height of 32px
- Large buttons should have a default height of 40px

image

image

image

## New Behavior

`Buttons` now have the default height specified in their design specs:

image

image

image

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                          |     Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                                   Change |
| :--------------------------------------------------------------------------------------------------------- | ---------------------------: | ---------------------------: | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-alert</samp> <br /> <abbr title='bundle-size/Alert.fixture.js'>Alert</abbr>                    | `86.163 kB`<br />`21.727 kB` | `86.691 kB`<br />`21.879 kB` | `528 B` <br />`152 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/Button.fixture.js'>Button</abbr>                 |  `33.487 kB`<br />`8.822 kB` |  `34.015 kB`<br />`8.986 kB` | `528 B` <br />`164 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/CompoundButton.fixture.js'>CompoundButton</abbr> | `40.507 kB`<br />`10.172 kB` | `41.035 kB`<br />`10.339 kB` | `528 B` <br />`167 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/MenuButton.fixture.js'>MenuButton</abbr>         | `38.167 kB`<br />`10.153 kB` | `38.695 kB`<br />`10.307 kB` | `528 B` <br />`154 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/SplitButton.fixture.js'>SplitButton</abbr>       | `45.737 kB`<br />`11.577 kB` | `46.265 kB`<br />`11.736 kB` | `528 B` <br />`159 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/ToggleButton.fixture.js'>ToggleButton</abbr>     |  `49.45 kB`<br />`10.849 kB` | `49.978 kB`<br />`11.002 kB` | `528 B` <br />`153 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                  |         Size (minified/GZIP) |
| -------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.All.fixture.js'>Card - All</abbr>     | `78.082 kB`<br />`22.521 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.fixture.js'>Card</abbr>               |  `72.92 kB`<br />`21.108 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardFooter.fixture.js'>CardFooter</abbr>   |    `9.04 kB`<br />`3.807 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardHeader.fixture.js'>CardHeader</abbr>   |  `10.916 kB`<br />`4.487 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardPreview.fixture.js'>CardPreview</abbr> |   `9.847 kB`<br />`4.159 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/21a34685cddbbae6531152e9f3393e237ba7b92c'>21a34685cddbbae6531152e9f3393e237ba7b92c</a></sub>
<div id="bundleSizeComment9423"/>

**Review by micahgodbolt — APPROVED** [review]

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26522/perf-test-react-components/Avatar-mount_base.html">1292</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26522/perf-test-react-components/Avatar-mount.html">1277</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26522/perf-test-react-components/Button-mount_base.html">927</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26522/perf-test-react-components/Button-mount.html">932</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26522/perf-test-react-components/FluentProvider-mount_base.html">1548</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26522/perf-test-react-components/FluentProvider-mount.html">1560</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26522/perf-test-react-components/FluentProviderWithTheme-mount_base.html">587</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26522/perf-test-react-components/FluentProviderWithTheme-mount.html">582</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26522/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">542</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26522/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">554</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26522/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">587</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26522/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">573</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26522/perf-test-react-components/InfoButton-mount_base.html">515</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26522/perf-test-react-components/InfoButton-mount.html">514</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26522/perf-test-react-components/MakeStyles-mount_base.html">2003</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26522/perf-test-react-components/MakeStyles-mount.html">1951</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26522/perf-test-react-components/Persona-mount_base.html">2850</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26522/perf-test-react-components/Persona-mount.html">2814</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26522/perf-test-react-components/SpinButton-mount_base.html">2357</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26522/perf-test-react-components/SpinButton-mount.html">2365</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Review by spmonahan — APPROVED** [review]

File: change/@fluentui-react-button-2cbb912a-03db-43b4-b4b5-9689954351b0.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-button-2cbb912a-03db-43b4-b4b5-9689954351b0.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the 73 screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-fluent/index.html?organization=uifabric&project=fabricpublic&prId=26522&commitId=fff7e239d6e29ebaab7f5f27f9c797c122b118d9&env=prod)

✅ There was 0 screenshots added, 0 screenshots removed, 1796 screenshots unchanged, 0 screenshots with different dimensions and 73 screenshots with visible difference.


  <details><summary><b>unknown</b> 73 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Button Converged.Circular and icon only - RTL.default.chromium.png|69| Changed|
|Button Converged.Circular and icon only - RTL.hover.chromium.png|50| Changed|
|Button Converged.Circular and icon only - RTL.pressed.chromium.png|51| Changed|
|Button Converged.Circular and icon only.default.chromium.png|69| Changed|
|Button Converged.Circular and icon only.hover.chromium.png|50| Changed|
|Button Converged.Circular and icon only.pressed.chromium.png|51| Changed|
|Button Converged.Icon only.default.chromium.png|68| Changed|
|Button Converged.Icon only.hover.chromium.png|48| Changed|
|Button Converged.Icon only.pressed.chromium.png|50| Changed|
|Button Converged.Size large - with long text wrapping.chromium.png|1614| Changed|
|Button Converged.Size large.default.chromium.png|263| Changed|
|Button Converged.Size large.hover.chromium.png|245| Changed|
|Button Converged.Size large.pressed.chromium.png|256| Changed|
|Button Converged.Size medium - with long text wrapping.chromium.png|861| Changed|
|Button Converged.Size small - with long text wrapping.chromium.png|875| Changed|
|Button Converged.Size small.default.chromium.png|248| Changed|
|Button Converged.Size small.hover.chromium.png|240| Changed|
|Button Converged.Size small.pressed.chromium.png|266| Changed|
|Card Converged.CardHeader.normal.chromium.png|4026| Changed|
|CompoundButton Converged.Size small - with long text wrapping.chromium.png|2| Changed|
|CompoundButton Converged.Size small.default.chromium.png|2| Changed|
|CompoundButton Converged.Size small.hover.chromium.png|2| Changed|
|CompoundButton Converged.Size small.pressed.chromium.png|2| Changed|
|Dialog.title custom action - Dark Mode.chromium.png|3215| Changed|
|Dialog.title custom action - High Contrast.chromium.png|5218| Changed|
|Dialog.title custom action - RTL.chromium.png|3478| Changed|
|Dialog.title custom action.chromium.png|3485| Changed|
|MenuButton Converged.Circular and icon only.default.chromium.png|69| Changed|
|MenuButton Converged.Circular and icon only.hover.chromium.png|50| Changed|
|MenuButton Converged.Circular and icon only.pressed.chromium.png|51| Changed|
|MenuButton Converged.Icon only.default.chromium.png|68| Changed|
|MenuButton Converged.Icon only.hover.chromium.png|48| Changed|
|MenuButton Converged.Icon only.pressed.chromium.png|50| Changed|
|MenuButton Converged.Size large - with long text wrapping.chromium.png|1635| Changed|
|MenuButton Converged.Size large.default.chromium.png|284| Changed|
|MenuButton Converged.Size large.hover.chromium.png|266| Changed|
|MenuButton Converged.Size large.pressed.chromium.png|280| Changed|
|MenuButton Converged.Size medium - with long text wrapping.chromium.png|870| Changed|
|MenuButton Converged.Size small - with long text wrapping.chromium.png|888| Changed|
|MenuButton Converged.Size small.default.chromium.png|265| Changed|
|MenuButton Converged.Size small.hover.chromium.png|258| Changed|
|MenuButton Converged.Size small.pressed.chromium.png|289| Changed|
|ToggleButton Converged.Circular and icon only.default.chromium.png|69| Changed|
|ToggleButton Converged.Circular and icon only.hover.chromium.png|50| Changed|
|ToggleButton Converged.Circular and icon only.pressed.chromium.png|51| Changed|
|ToggleButton Converged.Icon only.default.chromium.png|68| Changed|
|ToggleButton Converged.Icon only.hover.chromium.png|48| Changed|
|ToggleButton Converged.Icon only.pressed.chromium.png|50| Changed|
|ToggleButton Converged.Size large - with long text wrapping.chromium.png|1614| Changed|
|ToggleButton Converged.Size large.default.chromium.png|263| Changed|
|ToggleButton Converged.Size large.hover.chromium.png|245| Changed|
|ToggleButton Converged.Size large.pressed.chromium.png|256| Changed|
|ToggleButton Converged.Size medium - with long text wrapping.chromium.png|861| Changed|
|ToggleButton Converged.Size small - with long text wrapping.chromium.png|875| Changed|
|ToggleButton Converged.Size small.default.chromium.png|248| Changed|
|ToggleButton Converged.Size small.hover.chromium.png|240| Changed|
|ToggleButton Converged.Size small.pressed.chromium.png|266| Changed|
|Toolbar Converged.Default.Button Pressed.chromium.png|68| Changed|
|Toolbar Converged.Default.Toggle On.chromium.png|64| Changed|
|Toolbar Converged.Default.default.chromium.png|62| Changed|
|Toolbar Converged.Far Group.chromium.png|186| Changed|
|Toolbar Converged.Large.Button Pressed.chromium.png|72| Changed|
|Toolbar Converged.Large.Toggle On.chromium.png|71| Changed|
|Toolbar Converged.Large.default.chromium.png|68| Changed|
|Toolbar Converged.Small.Button Pressed.chromium.png|139| Changed|
|Toolbar Converged.Small.Toggle On.chromium.png|147| Changed|
|Toolbar Converged.Small.default.chromium.png|147| Changed|
|Toolbar Converged.Vertical Button.Button Pressed.chromium.png|53| Changed|
|Toolbar Converged.Vertical Button.Toggle On.chromium.png|43| Changed|
|Toolbar Converged.Vertical Button.default.chromium.png|43| Changed|
|Toolbar Converged.Vertical.Button Pressed.chromium.png|5052| Changed|
|Toolbar Converged.Vertical.Toggle On.chromium.png|5055| Changed|
|Toolbar Converged.Vertical.default.chromium.png|5062| Changed|

</details>


<div id="vrtCommentfluentuiv9"/>

File: packages/react-components/react-button/src/components/Button/useButtonStyles.ts [hunk 2]
```diff
@@ -395,19 +404,19 @@ const useRootFocusStyles = makeStyles({
 const useRootIconOnlyStyles = makeStyles({
   // Size variations
   small: {
-    ...shorthands.padding(tokens.spacingHorizontalXS),
+    ...shorthands.padding(buttonSpacingSmallWithIcon),
 
-    minWidth: '28px',
-    maxWidth: '28px',
+    minWidth: '24px',
+    maxWidth: '24px',
```
Comment (behowell) [inline packages/react-components/react-button/src/components/Button/useButtonStyles.ts:410]: nit: Are these minWidth/maxWidth still necessary? The button should be the right size based on the icon size plus padding and border, right?

**Review by behowell — APPROVED** [review]

File: packages/react-components/react-button/src/components/Button/useButtonStyles.ts [hunk 3]
```diff
@@ -13,7 +13,10 @@ export const buttonClassNames: SlotClassNames<ButtonSlots> = {
 const iconSpacingVar = '--fui-Button__icon--spacing';
 
 const buttonSpacingSmall = '3px';
+const buttonSpacingSmallWithIcon = '1px';
 const buttonSpacingMedium = '5px';
+const buttonSpacingLarge = tokens.spacingHorizontalS;
+const buttonSpacingLargeWithIcon = '7px';
```
[the remark this answers is not in the document]
Comment (khmakoto) [reply packages/react-components/react-button/src/components/Button/useButtonStyles.ts:19]: It's not necessarily that they want custom paddings, they want a specific height and the paddings are just the ones calculated to get that specific height.

File: packages/react-components/react-button/src/components/Button/useButtonStyles.ts (hunk 2 above)
Comment (khmakoto) [reply packages/react-components/react-button/src/components/Button/useButtonStyles.ts:410]: I'll check if I can get these without specifying minWidth/maxWidth in a separate PR to not delay the merging of this one.

File: change/@fluentui-react-button-2cbb912a-03db-43b4-b4b5-9689954351b0.json (hunk 1 above)
Comment (khmakoto) [reply change/@fluentui-react-button-2cbb912a-03db-43b4-b4b5-9689954351b0.json:1]: All changed screenshots are intended.

**Pull request merged**